test/03_device-state.ts

```diff
@@ -510,6 +555,316 @@ export default () => {
+										});
+
+										// Wait for the stale (TIMEOUT -> OFFLINE) message queue item to get consumed.
+										await setTimeout(TIMEOUT_SEC * 1000 + 1000);
+
+										// The write cache should then agree with the DB on the device being Online (but atm this fails)
+										// TODO: This should be fixed to be either offline or undefined.
+										expect(
+											await getHeartbeatWriteCacheState(device3.id),
+										).to.have.property('currentState', 'online');
+										await expectResourceToMatch(
+											pineUser,
+											'device',
+											device3.id,
+											{
+												// We use the same description as in the itExpectsError, so that we are sure which part failed.
+												api_heartbeat_state: (prop, value) =>
+													prop.to.equal(
+														DeviceOnlineStates.Online,
+														`The api_heartbeat_state should have become online but it was found ${value}`,
+													),
+											},
+										);
+									},
+									/The api_heartbeat_state should have become online but it was found offline/,
+								);
+
+								itExpectsError(
+									'should switch to Online after a subsequent state GET and not stay stuck as offline forever',
+									async function () {
+										// Confirm that we are int he case that the DB says Offline & the write Cache says Online
+										await expectDeviceHeartbeat(device3.id, {
+											db: DeviceOnlineStates.Offline,
+											cache: DeviceOnlineStates.Online,
+										});
+
+										await fakeDevice.getState(
+											device3,
+											device3.uuid,
+											stateVersion,
+										);
+										await setTimeout(maxGetStateEventConsumptionTimeout);
+										// B/c the write cache's state was already online, there was no device PATCH
+										expect(device3ChangeEventSpy.called).to.be.false;
+										await expectDeviceHeartbeat(device3.id, {
+											db: DeviceOnlineStates.Offline,
+											cache: DeviceOnlineStates.Online,
+										});
+
+										await expectResourceToMatch(
+											pineUser,
+											'device',
+											device3.id,
+											{
+												// We use the same description as in the itExpectsError, so that we are sure which part failed.
+												api_heartbeat_state: (prop, value) =>
+													prop.to.equal(
+														DeviceOnlineStates.Online,
+														`The api_heartbeat_state should have recovered to online but it was found ${value}`,
+													),
+											},
+										);
+										expect(
+											await getHeartbeatWriteCacheState(device3.id),
+										).to.have.property('currentState', 'online');
+									},
+									/The api_heartbeat_state should have recovered to online but it was found offline/,
+								);
+
+								// TODO: This should not be happening. Drop this v test once the previous tests gets fixed.
+								itExpectsError(
+									'[FIXME] is expected that it will stay stuck to Offline when doing regular state GETs',
```

Comment:
This whole test case gets removed in #1560 since the device will now be switching to online on the first state GET, so if you see anything ugly, be sure that it's going away right after this gets merged.